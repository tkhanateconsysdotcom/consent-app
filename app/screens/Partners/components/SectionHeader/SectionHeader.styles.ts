import { StyleSheet } from "react-native";

import { Layout } from "constant";
import { Colors, ComponentsStyle } from "style";

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    paddingVertical: 10,
    paddingHorizontal: Layout.PADDING_HORIZONTAL,
    backgroundColor: Colors.white,
    ...ComponentsStyle.header,
  },
  text: {
    textTransform: "capitalize",
  },
});

export default styles;
