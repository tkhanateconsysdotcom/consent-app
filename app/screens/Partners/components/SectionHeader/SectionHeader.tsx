import React from "react";
import { View } from "react-native";
import moment from "moment";

import "moment/min/locales";
import { Text } from "components";
import { withLocalization, LocalizationContextProps, t } from "utils";

import styles from "./SectionHeader.styles";

interface Props {
  date: Date;
  co2value: number;
  monthlyAcceptanceConsent: number;
}

const SectionHeader = ({
  date,
  co2value = 1,
  monthlyAcceptanceConsent = 1,
  language = "",
}: Props & LocalizationContextProps) => {
  let percentageConsent = 1;

  if (monthlyAcceptanceConsent && co2value) {
    percentageConsent = Math.round((co2value / monthlyAcceptanceConsent) * 100);
    if (percentageConsent < 1) {
      percentageConsent = 1;
    }
  }

  return (
    <View style={styles.container}>
      <Text.Primary darkGray bold style={styles.text}>
        {moment(date).locale(language).format("MMMM YYYY")}
      </Text.Primary>
      {percentageConsent && (
        <Text.Secondary
          red={percentageConsent > 100}
          green={percentageConsent < 100}
        >
          {percentageConsent + " % " + t("PARTNERS_SCREEN_HEADER_OF_CONSENT")}
        </Text.Secondary>
      )}
    </View>
  );
};

export default withLocalization(SectionHeader);
