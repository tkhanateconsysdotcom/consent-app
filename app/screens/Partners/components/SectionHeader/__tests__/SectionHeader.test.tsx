import React from "react";
import { create } from "react-test-renderer";

import SectionHeader from "../SectionHeader";

jest.unmock("../SectionHeader");

describe("SectionHeader", () => {
  const date = new Date("2020-12-24T03:24:00");

  it("should render when user is over consent", () => {
    const tree = create(
      <SectionHeader co2value={30} monthlyAcceptanceConsent={20} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when user is below consent", () => {
    const tree = create(
      <SectionHeader co2value={10} monthlyAcceptanceConsent={20} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when user has consent below 1%", () => {
    const tree = create(
      <SectionHeader co2value={1} monthlyAcceptanceConsent={1000} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when co2value equals 0", () => {
    const tree = create(
      <SectionHeader co2value={0} monthlyAcceptanceConsent={1} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when monthlyAcceptanceConsent equals 0", () => {
    const tree = create(
      <SectionHeader co2value={1} monthlyAcceptanceConsent={0} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when monthlyAcceptanceConsent and co2value equals 0", () => {
    const tree = create(
      <SectionHeader co2value={0} monthlyAcceptanceConsent={0} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
