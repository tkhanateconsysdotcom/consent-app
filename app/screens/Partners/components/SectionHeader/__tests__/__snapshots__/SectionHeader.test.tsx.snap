// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader should render when co2value equals 0 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 14,
      "paddingVertical": 10,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <Text.Secondary
    green={true}
    red={false}
  >
    1 % of consent
  </Text.Secondary>
</View>
`;

exports[`SectionHeader should render when monthlyAcceptanceConsent and co2value equals 0 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 14,
      "paddingVertical": 10,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <Text.Secondary
    green={true}
    red={false}
  >
    1 % of consent
  </Text.Secondary>
</View>
`;

exports[`SectionHeader should render when monthlyAcceptanceConsent equals 0 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 14,
      "paddingVertical": 10,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <Text.Secondary
    green={true}
    red={false}
  >
    1 % of consent
  </Text.Secondary>
</View>
`;

exports[`SectionHeader should render when user has consent below 1% 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 14,
      "paddingVertical": 10,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <Text.Secondary
    green={true}
    red={false}
  >
    1 % of consent
  </Text.Secondary>
</View>
`;

exports[`SectionHeader should render when user is below consent 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 14,
      "paddingVertical": 10,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <Text.Secondary
    green={true}
    red={false}
  >
    50 % of consent
  </Text.Secondary>
</View>
`;

exports[`SectionHeader should render when user is over consent 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 14,
      "paddingVertical": 10,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <Text.Secondary
    green={false}
    red={true}
  >
    150 % of consent
  </Text.Secondary>
</View>
`;
